swap_meet/vendor.py

```diff
@@ -1,2 +1,83 @@
 class Vendor:
-    pass
\ No newline at end of file
+    def __init__(self, inventory=None):
+        if not inventory:
+            inventory = []
+        self.inventory = inventory
+
+    def add(self, item):
+        # Add an item to inventory
+        if item:
+            self.inventory.append(item)
+        return item
+
+    def remove(self, item):
+        # Remove an item from inventory
+        try:
+            self.inventory.remove(item)
+            return item
+        except ValueError as err:
+            print(f"{err}. Unable to remove '{item}' from inventory.")
+            return False
+
+    def get_by_id(self, id):
+        # Return item that matches target id
+        try:
+            for item in self.inventory:
+                if item.id == id:
+                    return item
+        except IndexError as err:
+            print(f"{err}. Could not retrieve item '{id}' from inventory.")
+            return None
+
+    def swap_items(self, other_vendor, my_item, their_item):
+        # Check if swap is even possible
+        if (my_item not in self.inventory or their_item not in other_vendor.inventory):
+            return False
+
+        self.remove(my_item)
+        other_vendor.add(my_item)
+        other_vendor.remove(their_item)
+        self.add(their_item)
+        return True
+
+    def swap_first_item(self, other_vendor):
+        # Get first item in each inventory and swap them
+        try:
+            return self.swap_items(other_vendor, self.inventory[0], other_vendor.inventory[0])
+        except IndexError as err:
+            print(f"{err}. Item does not exist.")
```

Comment:
An IndexError would be caused by either inventory being empty and the index `0` not existing. This error message isn't quite accurate.